src/plugins/intel_gpu/src/graph/layout_optimizer.cpp

```diff
@@ -241,13 +241,37 @@ bool layout_optimizer::can_fuse_reorder(program_node& prev, program_node& next,
     if (next.is_type<reorder>())
         return true;
 
-    // Errata for onednn layout selection
-    if (next.is_type<convolution>() && next.get_dependencies().size() == 1 &&
+    // Check whether the reorder between prev and next is the first input of next.
+    auto is_input_reorder = [](program_node& prev, program_node& next) {
+        auto found_reorder = std::find_if(next.get_dependencies().begin(), next.get_dependencies().end(), [](cldnn::program_node* node){
+            return node->is_type<reorder>();
+        });
+        // if there is no reorder between prev and next, it returns true.
+        // This case is needed for can_fuse_reorder in reorder_inputs pass.
+        if (found_reorder == std::end(next.get_dependencies())) {
```

Comment:
I think this should be `if (found_reorder == std::end(next.get_dependencies()) && next.get_dependency_index(prev) == 0)` as we are handling the first input.